fractions/jose/src/main/java/org/wildfly/swarm/jose/DecryptionOutput.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.jose;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class DecryptionOutput {
+    private Map<String, Object> metadata;
+    private String data;
+    public DecryptionOutput(String data) {
+        this(Collections.emptyMap(), data);
+    }
+    public DecryptionOutput(Map<String, Object> metadata, String data) {
+        this.metadata = metadata;
+        this.data = data;
+    }
+    public Map<String, Object> getMetadata() {
```

Comment:
Is the `Map` that callers can access through this method expected to be modified? If so, probably shouldn't use `Collections.emptyMap()` in the 1st constructor.

Reply:
No, on the output it is not modifiable, these are the headers which came with the signed/encryped payload, they can contain the crypto related infor but also custom values, I will add the docs. And rename 'metadata' to 'headers' because this is what it is :-). On the input, it is the way to  customize the process further and also pass along the custom values - since the headers are also protected.